### ISSUE TITLE
Channel size & number of resources reported incorrectly after the first import is cancelled

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#4409]
### Observed behavior

When the user selects the channel to import, and imports just the DB (_Generating the channel listing..._), but then closes the _Select content from..._ window and selects the same channel again, its size & number of resources are reported as 0. Happens on both import from local drive and Studio.

Once resources are selected, the import proceeds without issues, and on subsequent _Import more_ tries, the size is reported correctly.

![ubuntu 16 04 3 start running - oracle vm virtualbox_014](https://user-images.githubusercontent.com/1457929/46892460-c6fc9800-ce6d-11e8-91ac-ee6001169151.png)

### Expected behavior
Channel size & number of resources should be reported correctly even after cancelling the previous import attempts.

### User-facing consequences
Incorrect information and uncertainty about the channel to be imported.

#### Browser console
![ubuntu 16 04 3 start running - oracle vm virtualbox_016](https://user-images.githubusercontent.com/1457929/46892865-33c46200-ce6f-11e8-8e04-10b64bb74024.png)

Not seeing any errors in the `server.log`
```
INFO:cherrypy.access.140653758354152:127.0.0.1 - - [12/Oct/2018:21:13:21] "GET /static/device_management_module/device_management_module-0.11.0b1.js HTTP/1.1" 304 - "http://127.0.0.1:8080/device/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0"
[37mINFO     Downloading data for channel id 000409f81dbe5d1ba67101cb9fed4530[0m
INFO:kolibri.core.content.management.commands.importchannel:Downloading data for channel id 000409f81dbe5d1ba67101cb9fed4530

  0%|          | 0/1343488 [00:00<?, ?it/s]
100%|██████████| 1343488/1343488 [00:00<00:00, 1653905.15it/s][37mINFO     Older version 5 of channel 000409f81dbe5d1ba67101cb9fed4530 already exists in database; removing old entries so we can upgrade to version 6[0m
INFO:kolibri.core.content.utils.channel_import:Older version 5 of channel 000409f81dbe5d1ba67101cb9fed4530 already exists in database; removing old entries so we can upgrade to version 6
[37mINFO     Importing ContentTag data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentTag data
[37mINFO     Importing ContentNode_tags data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_tags data
[37mINFO     Importing ContentNode_related data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_related data
[37mINFO     Importing ContentNode_has_prerequisite data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_has_prerequisite data
[37mINFO     Importing ContentNode data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode data
[37mINFO     Importing Language data[0m
INFO:kolibri.core.content.utils.channel_import:Importing Language data
[37mINFO     Importing File data[0m
INFO:kolibri.core.content.utils.channel_import:Importing File data
[37mINFO     Importing LocalFile data[0m
INFO:kolibri.core.content.utils.channel_import:Importing LocalFile data
[37mINFO     Importing AssessmentMetaData data[0m
INFO:kolibri.core.content.utils.channel_import:Importing AssessmentMetaData data
[37mINFO     Importing ChannelMetadata data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ChannelMetadata data
[37mINFO     Setting availability of File objects based on LocalFile availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of File objects based on LocalFile availability
[37mINFO     Setting availability of non-topic ContentNode objects based on File availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of non-topic ContentNode objects based on File availability
[37mINFO     Setting availability of ContentNode objects with children for 5 levels[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for 5 levels
[37mINFO     Setting availability of ContentNode objects with children for level 5[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 5
[37mINFO     Setting availability of ContentNode objects with children for level 4[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 4
[37mINFO     Setting availability of ContentNode objects with children for level 3[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 3
[37mINFO     Setting availability of ContentNode objects with children for level 2[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 2
[37mINFO     Setting availability of ContentNode objects with children for level 1[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 1

[37mINFO     Downloading data for channel id 0e173fca6e9052f8a474a2fb84055faf[0m
INFO:kolibri.core.content.management.commands.importchannel:Downloading data for channel id 0e173fca6e9052f8a474a2fb84055faf

  0%|          | 0/1534976 [00:00<?, ?it/s]
100%|██████████| 1534976/1534976 [00:00<00:00, 1868874.27it/s][37mINFO     Importing ContentTag data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentTag data
[37mINFO     Importing ContentNode_tags data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_tags data
[37mINFO     Importing ContentNode_related data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_related data
[37mINFO     Importing ContentNode_has_prerequisite data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_has_prerequisite data
[37mINFO     Importing ContentNode data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode data
[37mINFO     Importing Language data[0m
INFO:kolibri.core.content.utils.channel_import:Importing Language data
[37mINFO     Importing File data[0m
INFO:kolibri.core.content.utils.channel_import:Importing File data
[37mINFO     Importing LocalFile data[0m
INFO:kolibri.core.content.utils.channel_import:Importing LocalFile data
[37mINFO     Importing AssessmentMetaData data[0m
INFO:kolibri.core.content.utils.channel_import:Importing AssessmentMetaData data
[37mINFO     Importing ChannelMetadata data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ChannelMetadata data
[37mINFO     Setting availability of File objects based on LocalFile availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of File objects based on LocalFile availability
[37mINFO     Setting availability of non-topic ContentNode objects based on File availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of non-topic ContentNode objects based on File availability
[37mINFO     Setting availability of ContentNode objects with children for 5 levels[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for 5 levels
[37mINFO     Setting availability of ContentNode objects with children for level 5[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 5
[37mINFO     Setting availability of ContentNode objects with children for level 4[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 4
[37mINFO     Setting availability of ContentNode objects with children for level 3[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 3
[37mINFO     Setting availability of ContentNode objects with children for level 2[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 2
[37mINFO     Setting availability of ContentNode objects with children for level 1[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 1
```

### Steps to reproduce
1. Select a new channel to import
2. When the channel content tree loads, close the  _Select content from..._ window without actually importing anything
3. Select the same channel again
4. Observe the total size and number of resources values

### Context

 * Kolibri version: `0.11.0b1`
 * Operating system: Ubuntu 16.04
 * Browser: both FF & Chromium

cc @indirectlylit @rtibbles

**Comment by radinamatic** [comment]
Not sure if it's connected, but when trying to export to local drive, total size & number of resources in the channel is reported as 0, together with the number of resources already imported on the device, while their size is apparently reported correctly.

![ubuntu 16 04 3 start running - oracle vm virtualbox_024](https://user-images.githubusercontent.com/1457929/46901255-87a36b00-ceb0-11e8-9de2-fd9daf82ef95.png)

**Comment by mrpau-richard** [comment]
I can reproduced this issue on Windows 7 and 10 64 bit.

**Comment by jonboiser** [comment]
This might be a duplicate of #4331. We investigated it a bit and found that the RemoteChannel API endpoint that should be providing those resource/size numbers is return the correct data, but my theory is it might be reset somehow getting to the view.

**Comment by indirectlylit** [comment]
@jonboiser would you mind tackling this?

**Comment by indirectlylit** [comment]
note - does not seem to be fixed by #4437:

![import](https://user-images.githubusercontent.com/2367265/47057877-6fb13d00-d177-11e8-9883-515378ea7b11.gif)

**Comment by jonboiser** [comment]
The file/resource sizes when importing from Kolibri Studio should be fixed in the 0.11-release branch.

The file/resource size discrepancy in Export _is_ an issue that seems to have stemmed from some of the speed up work in #3885. Need to make sure that the full size statistics are available at parts of the workflow that require it. Previously, we relied on all the file/resource stats being available because they came as a group (now they can be selected as needed)

**Comment by indirectlylit** [comment]
> The file/resource sizes when importing from Kolibri Studio should be fixed in the 0.11-release branch.

confirmed

**Comment by indirectlylit** [comment]
closing and migrating other issue to #4452
